Create models

Ochiengsteven opened the pull request:
## Pull Request: Project Initialization and Database Setup💽
### Changes Made

- [x] New app created and named appropriately.
- [x] Git initialized for version control. 🌳
- [x] Postgres database set up. 🔄🐘
- [x] Project schema built according to ERD diagram.
- [x] Migration files created and executed successfully. ⚙️
- [x] Table and column names aligned with specifications.
- [x] 'photo' column in 'users' table linked to an image. 📸
- [x] Foreign key column in 'posts' table named 'author_id'. 🤝
- [x] Foreign keys included with corresponding indexes. 📊
- [x] References from sneak peek followed for clarity.

### Additional Notes

- Followed the guidelines and references from the sneak peek to ensure accurate table and column names.

### Testing

1. Manually verified that the database schema matches the expected structure.






Please review and allow merge.

Thank you! 🙌

Ochiengsteven committed:
ADD RUBOCOP

Ochiengsteven committed:
ADD STYLELINT

Ochiengsteven committed:
CREATE TABLES AND RELATIONSHIPS

Ochiengsteven committed:
ADD README

Ochiengsteven committed:
ADD LICENSE

Ibizugbe commented on db/migrate/20240106165245_create_posts.rb:
```diff
@@ -0,0 +1,14 @@
+class CreatePosts < ActiveRecord::Migration[7.1]
+  def change
+    create_table :posts do |t|
+      t.integer :author_id
+      t.string :title
+      t.text :text
+      t.integer :likes_counter, default: 0
+      t.integer :comments_counter, default: 0
+
+      t.timestamps
+    end
+    add_index :posts, :author_id
```
Nice job so far👍🏼. Just as a requirement, kindly do well to add the necessary foreign keys as shown in the [ERD ](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_app_erd_v1_1.png)😉.

<img width="962" alt="image" src="https://github.com/Ochiengsteven/art-post/assets/64643995/2aac4674-e684-4c01-bfcb-e49bcd62330d">

Kindly add the necessary foreign keys for the post, likes, and comments😉.

For instance, we can make some improvements here in the post table by adding foreign keys as shown below.

```suggestion
  def change
    create_table :posts do |t|
      t.integer :author_id
      t.string :title
      t.text :text
      t.integer :likes_counter, default: 0
      t.integer :comments_counter, default: 0

      t.timestamps
    end
    add_foreign_key :posts, :users, column: :author_id
    add_index :posts, :author_id
```

Kindly note that the above is a suggestion and you should try to implement it in any way that works😉

Ibizugbe reviewed (changes requested):
Hi @Ochiengsteven 👋🏼👋🏼👋🏼, 

![request change](https://github.com/Ochiengsteven/art-post/assets/64643995/2dd960f9-7b63-4cf5-86b8-821ae7032a09)


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] Linters are passing👍🏼.
- [x] Database adapter is using Postgresq👍🏼.


## Required Changes ♻️

- [ ] Kindly add the necessary foreign keys for the post, likes, and comments table as I have suggested in the comment below😉.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Ochiengsteven committed:
add foreign keys

TedLivist reviewed (approved):
## STATUS: APPROVED ✔️ ✔️

Hi @Ochiengsteven,

✔️ Use of PostgreSQL for database

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Ochiengsteven commented:
@Ibizugbe Thanks for the review. I've made the following updates as requested:
In the migrations:

For the posts table, I added a foreign key constraint on the `author_id` column, referencing the `users` table.
For the `likes `and `comments` tables, I used `references` for the `user` and `post` columns, which **implicitly adds foreign key constraints**.
Additionally, I added an index on the `likes` table for the combination of `user_id` and `post_id` to ensure uniqueness, and I added an index on the `comments` table for the same combination.
